scripts/fetch_tools.sh

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS=$'\n\t'
+
+ARCH="$(uname -p)"
+DEST_DIR="${DEST_DIR:-_output/bin}"
+
+_download() {
+    local output="$1"
+    local url="$2"
+    for _ in $(seq 1 5); do
+        if curl -sSfLO --output-dir "${output}" "${url}"; then
+            return 0
+        fi
+        sleep 5s
+    done
+    return 1
+}
+
+_install() {
+    local url="$1"
+    local checksum="$2"
+    local filename="$3"
+    local initial_filename="$4"
+    local dest="$DEST_DIR/$filename"
+
+    [[ -e "${dest}" ]] && return 0
+    echo "Installing $filename to $DEST_DIR"
+
+    tmp=$(mktemp -d)
+    trap 'rm -rfv ${tmp} &>/dev/null' EXIT
+
+    filename="$(basename "$url")"
+    echo -n "$checksum -" >"$tmp/checksum.txt"
+
+    _download "$tmp" "$url" 
+
+    if ! sha256sum -c "${tmp}/checksum.txt" < "$tmp/$filename" &>/dev/null; then
+        echo "  Checksum for $filename doesn't match"
+        echo "    Expected: $checksum"
+        echo "         Got: $(sha256sum < "$tmp/$filename" | cut -d' ' -f1)"
+        return 1
+    fi
+
+    if [[ "$(file --brief --mime-type "$tmp/$filename")" != "application/x-executable" ]]; then
+        (cd "$tmp" && tar xvf "$filename" --transform 's,.*\/,,g' --wildcards "*/$initial_filename" >/dev/null)
+    fi
```

Comment:
What are we doing here? Please, also add a comment in the code before the "if" block

Reply:
added comment